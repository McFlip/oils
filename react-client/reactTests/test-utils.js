import { render } from 'react-testing-library';
import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import promise from "redux-promise";
import reducers from 'reducers/';
import { Provider } from 'react-redux';
import { BrowserRouter } from "react-router-dom";

function renderWithRedux(
  ui,
  {initialState, reducer = (state = {}) => state, store = createStore(reducer, initialState)} = {}) {
  return {
    ...render(
      <Provider store={store}>
        <BrowserRouter>
          {ui}
        </BrowserRouter>
      </Provider>
    ),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  }
}

export const newStore = () => {
  const createStoreWithMiddleware = applyMiddleware(promise)(createStore);
  const store = createStoreWithMiddleware(reducers, {});
  return store;
}


// re-export everything
export * from 'react-testing-library';

// override render method
export {renderWithRedux as render};
