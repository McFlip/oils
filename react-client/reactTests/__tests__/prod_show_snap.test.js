/* eslint-env jest */
import React from 'react'
import { render, newStore } from '../test-utils'

import ProdsShow from 'components/prods_show'

jest.mock('actions/prods')

const store = newStore()

// check render
test('render product', () => {
  const product = render(
    <ProdsShow
      match={{
        params: { id: 'a' },
        isExact: true,
        path: '',
        url: ''
      }}
      history={{
        location: { state: undefined },
        push: () => null
      }}
    />,
    { store }
  )
  expect(product).toMatchSnapshot()
})
