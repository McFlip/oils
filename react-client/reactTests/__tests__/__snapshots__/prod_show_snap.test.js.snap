// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render product 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="navbar navbar-expand-sm bg-dark navbar-dark fixed-top"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            <img
              alt="Home"
              src="/favicon.ico"
            />
          </a>
          <button
            class="navbar-toggler"
            data-target="#navbarNavDropdown"
            data-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
            id="navbarNavDropdown"
          >
            <ul
              class="navbar-nav"
            >
              <li
                class="nav-item active"
              >
                <a
                  class="nav-link"
                  href="/products"
                >
                  products
                </a>
              </li>
              <li
                class="nav-item "
              >
                <a
                  class="nav-link"
                  href="/uses"
                >
                  uses
                </a>
              </li>
              <li
                class="nav-item "
              >
                <a
                  class="nav-link"
                  href="/recipes"
                >
                  recipes
                </a>
              </li>
              <li
                class="nav-item dropdown"
              >
                <a
                  class="nav-link dropdown-toggle"
                  data-toggle="dropdown"
                  href="#"
                  id="navbarDropdownMenuLink"
                  role="button"
                >
                  Actions
                </a>
                <div
                  class="dropdown-menu"
                >
                  <a
                    class="dropdown-item"
                    href="#"
                  >
                    Delete Product
                  </a>
                  <a
                    class="dropdown-item"
                    href="/products/a/edit"
                  >
                    Edit Product
                  </a>
                  <a
                    class="dropdown-item"
                    href="/products/a/editingredients"
                  >
                    Edit Contents
                  </a>
                  <a
                    class="dropdown-item"
                    href="/products/a/adduse"
                  >
                    Add Use
                  </a>
                  <a
                    class="dropdown-item"
                    href="/products/a/addpost"
                  >
                    Add Post
                  </a>
                </div>
              </li>
            </ul>
          </div>
        </nav>
        <div
          class="jumbotron"
        >
          <h1>
            test 1 oz 
          </h1>
          <span
            class="form-group form-check-inline px-2"
          >
            <label
              class="form-check-label"
              for="checkbox_a"
            >
               wishlist 
            </label>
            <input
              class="form-check-input ml-2"
              id="checkbox_a"
              type="checkbox"
            />
          </span>
          <h2>
            Category: 
          </h2>
          
          <div
            class="form-group"
          >
            <div
              class="form-check"
            >
              <input
                checked=""
                class="form-check-input"
                id="photosensitive"
                readonly=""
                type="checkbox"
              />
              <label
                for="photosensitive"
              >
                photosensitive
              </label>
            </div>
            <div
              class="form-check"
            >
              <input
                checked=""
                class="form-check-input"
                id="topical"
                readonly=""
                type="checkbox"
              />
              <label
                for="topical"
              >
                topical
              </label>
            </div>
            <div
              class="form-check"
            >
              <input
                checked=""
                class="form-check-input"
                id="dilute"
                readonly=""
                type="checkbox"
              />
              <label
                for="dilute"
              >
                dilute
              </label>
            </div>
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="aromatic"
                readonly=""
                type="checkbox"
              />
              <label
                for="aromatic"
              >
                aromatic
              </label>
            </div>
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="dietary"
                readonly=""
                type="checkbox"
              />
              <label
                for="dietary"
              >
                dietary
              </label>
            </div>
          </div>
        </div>
        <div
          id="accordian"
        >
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              <a
                class="card-link"
                data-toggle="collapse"
                href="#Contains"
              >
                <h4>
                  Contains
                  :
                </h4>
              </a>
            </div>
            <div
              class="collapse"
              data-parent="#accordian"
              id="Contains"
            >
              <div>
                <ul
                  class="list-group"
                >
                  <li
                    class="list-group-item input-group"
                  >
                    <div>
                      <h4
                        class="alert alert-yl shadow-lg"
                      >
                        content category
                      </h4>
                      <hr />
                    </div>
                    <span
                      data-testid="sku"
                    />
                    <span
                      class="px-2"
                    >
                      <a
                        data-testid="descr"
                        href="/products/contentId"
                      >
                        content 1
                      </a>
                    </span>
                    <span
                      class="px-2"
                      data-testid="size"
                    />
                    
                    <span
                      class="form-group form-check-inline px-2"
                    >
                      <label
                        class="form-check-label"
                        for="checkbox_contentId"
                      >
                         wishlist 
                      </label>
                      <input
                        class="form-check-input ml-2"
                        id="checkbox_contentId"
                        type="checkbox"
                      />
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              <a
                class="card-link"
                data-toggle="collapse"
                href="#ContainedIn"
              >
                <h4>
                  Contained In
                  :
                </h4>
              </a>
            </div>
            <div
              class="collapse"
              data-parent="#accordian"
              id="ContainedIn"
            >
              <div>
                <ul
                  class="list-group"
                >
                  <li
                    class="list-group-item input-group"
                  >
                    <div>
                      <h4
                        class="alert alert-yl shadow-lg"
                      >
                        container category
                      </h4>
                      <hr />
                    </div>
                    <span
                      data-testid="sku"
                    />
                    <span
                      class="px-2"
                    >
                      <a
                        data-testid="descr"
                        href="/products/containerId"
                      >
                        container
                      </a>
                    </span>
                    <span
                      class="px-2"
                      data-testid="size"
                    />
                    
                    <span
                      class="form-group form-check-inline px-2"
                    >
                      <label
                        class="form-check-label"
                        for="checkbox_containerId"
                      >
                         wishlist 
                      </label>
                      <input
                        class="form-check-input ml-2"
                        id="checkbox_containerId"
                        type="checkbox"
                      />
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              <a
                class="card-link"
                data-toggle="collapse"
                href="#Uses"
              >
                <h4>
                  Uses
                  :
                </h4>
              </a>
            </div>
            <div
              class="collapse"
              data-parent="#accordian"
              id="Uses"
            >
              <div
                class="card"
              >
                <h4>
                  Uses:
                </h4>
                <ul
                  class="list-group"
                >
                  <li
                    class="list-group-item input-group"
                  >
                    <span
                      data-testid="use"
                    >
                      <a
                        data-testid="use"
                        href="/uses/undefined"
                      />
                    </span>
                    <span
                      class="px-2"
                    >
                      <button
                        class="btn btn-outline-danger float-right"
                        data-testid="use-btn"
                      >
                        Delete
                      </button>
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              <a
                class="card-link"
                data-toggle="collapse"
                href="#Posts"
              >
                <h4>
                  Posts
                  :
                </h4>
              </a>
            </div>
            <div
              class="collapse"
              data-parent="#accordian"
              id="Posts"
            >
              <div
                class="card"
              >
                <h4>
                  Posts:
                </h4>
                <div>
                  <div
                    class="card"
                  >
                    <img
                      alt="posted image"
                      class=""
                      src="http://localhost:3000/images/postImage"
                    />
                    <div
                      class="card-body"
                    >
                      <div
                        class="alert alert-yl"
                      >
                        postTitle
                        <a
                          class="btn float-right"
                          href="/products/a/editpost/postId"
                        >
                          ...
                        </a>
                      </div>
                      <div
                        class="card-text"
                      >
                        <p>
                          postContent
                        </p>
                        

                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="navbar navbar-expand-sm bg-dark navbar-dark fixed-top"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          <img
            alt="Home"
            src="/favicon.ico"
          />
        </a>
        <button
          class="navbar-toggler"
          data-target="#navbarNavDropdown"
          data-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarNavDropdown"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item active"
            >
              <a
                class="nav-link"
                href="/products"
              >
                products
              </a>
            </li>
            <li
              class="nav-item "
            >
              <a
                class="nav-link"
                href="/uses"
              >
                uses
              </a>
            </li>
            <li
              class="nav-item "
            >
              <a
                class="nav-link"
                href="/recipes"
              >
                recipes
              </a>
            </li>
            <li
              class="nav-item dropdown"
            >
              <a
                class="nav-link dropdown-toggle"
                data-toggle="dropdown"
                href="#"
                id="navbarDropdownMenuLink"
                role="button"
              >
                Actions
              </a>
              <div
                class="dropdown-menu"
              >
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Delete Product
                </a>
                <a
                  class="dropdown-item"
                  href="/products/a/edit"
                >
                  Edit Product
                </a>
                <a
                  class="dropdown-item"
                  href="/products/a/editingredients"
                >
                  Edit Contents
                </a>
                <a
                  class="dropdown-item"
                  href="/products/a/adduse"
                >
                  Add Use
                </a>
                <a
                  class="dropdown-item"
                  href="/products/a/addpost"
                >
                  Add Post
                </a>
              </div>
            </li>
          </ul>
        </div>
      </nav>
      <div
        class="jumbotron"
      >
        <h1>
          test 1 oz 
        </h1>
        <span
          class="form-group form-check-inline px-2"
        >
          <label
            class="form-check-label"
            for="checkbox_a"
          >
             wishlist 
          </label>
          <input
            class="form-check-input ml-2"
            id="checkbox_a"
            type="checkbox"
          />
        </span>
        <h2>
          Category: 
        </h2>
        
        <div
          class="form-group"
        >
          <div
            class="form-check"
          >
            <input
              checked=""
              class="form-check-input"
              id="photosensitive"
              readonly=""
              type="checkbox"
            />
            <label
              for="photosensitive"
            >
              photosensitive
            </label>
          </div>
          <div
            class="form-check"
          >
            <input
              checked=""
              class="form-check-input"
              id="topical"
              readonly=""
              type="checkbox"
            />
            <label
              for="topical"
            >
              topical
            </label>
          </div>
          <div
            class="form-check"
          >
            <input
              checked=""
              class="form-check-input"
              id="dilute"
              readonly=""
              type="checkbox"
            />
            <label
              for="dilute"
            >
              dilute
            </label>
          </div>
          <div
            class="form-check"
          >
            <input
              class="form-check-input"
              id="aromatic"
              readonly=""
              type="checkbox"
            />
            <label
              for="aromatic"
            >
              aromatic
            </label>
          </div>
          <div
            class="form-check"
          >
            <input
              class="form-check-input"
              id="dietary"
              readonly=""
              type="checkbox"
            />
            <label
              for="dietary"
            >
              dietary
            </label>
          </div>
        </div>
      </div>
      <div
        id="accordian"
      >
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <a
              class="card-link"
              data-toggle="collapse"
              href="#Contains"
            >
              <h4>
                Contains
                :
              </h4>
            </a>
          </div>
          <div
            class="collapse"
            data-parent="#accordian"
            id="Contains"
          >
            <div>
              <ul
                class="list-group"
              >
                <li
                  class="list-group-item input-group"
                >
                  <div>
                    <h4
                      class="alert alert-yl shadow-lg"
                    >
                      content category
                    </h4>
                    <hr />
                  </div>
                  <span
                    data-testid="sku"
                  />
                  <span
                    class="px-2"
                  >
                    <a
                      data-testid="descr"
                      href="/products/contentId"
                    >
                      content 1
                    </a>
                  </span>
                  <span
                    class="px-2"
                    data-testid="size"
                  />
                  
                  <span
                    class="form-group form-check-inline px-2"
                  >
                    <label
                      class="form-check-label"
                      for="checkbox_contentId"
                    >
                       wishlist 
                    </label>
                    <input
                      class="form-check-input ml-2"
                      id="checkbox_contentId"
                      type="checkbox"
                    />
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <a
              class="card-link"
              data-toggle="collapse"
              href="#ContainedIn"
            >
              <h4>
                Contained In
                :
              </h4>
            </a>
          </div>
          <div
            class="collapse"
            data-parent="#accordian"
            id="ContainedIn"
          >
            <div>
              <ul
                class="list-group"
              >
                <li
                  class="list-group-item input-group"
                >
                  <div>
                    <h4
                      class="alert alert-yl shadow-lg"
                    >
                      container category
                    </h4>
                    <hr />
                  </div>
                  <span
                    data-testid="sku"
                  />
                  <span
                    class="px-2"
                  >
                    <a
                      data-testid="descr"
                      href="/products/containerId"
                    >
                      container
                    </a>
                  </span>
                  <span
                    class="px-2"
                    data-testid="size"
                  />
                  
                  <span
                    class="form-group form-check-inline px-2"
                  >
                    <label
                      class="form-check-label"
                      for="checkbox_containerId"
                    >
                       wishlist 
                    </label>
                    <input
                      class="form-check-input ml-2"
                      id="checkbox_containerId"
                      type="checkbox"
                    />
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <a
              class="card-link"
              data-toggle="collapse"
              href="#Uses"
            >
              <h4>
                Uses
                :
              </h4>
            </a>
          </div>
          <div
            class="collapse"
            data-parent="#accordian"
            id="Uses"
          >
            <div
              class="card"
            >
              <h4>
                Uses:
              </h4>
              <ul
                class="list-group"
              >
                <li
                  class="list-group-item input-group"
                >
                  <span
                    data-testid="use"
                  >
                    <a
                      data-testid="use"
                      href="/uses/undefined"
                    />
                  </span>
                  <span
                    class="px-2"
                  >
                    <button
                      class="btn btn-outline-danger float-right"
                      data-testid="use-btn"
                    >
                      Delete
                    </button>
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
        
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <a
              class="card-link"
              data-toggle="collapse"
              href="#Posts"
            >
              <h4>
                Posts
                :
              </h4>
            </a>
          </div>
          <div
            class="collapse"
            data-parent="#accordian"
            id="Posts"
          >
            <div
              class="card"
            >
              <h4>
                Posts:
              </h4>
              <div>
                <div
                  class="card"
                >
                  <img
                    alt="posted image"
                    class=""
                    src="http://localhost:3000/images/postImage"
                  />
                  <div
                    class="card-body"
                  >
                    <div
                      class="alert alert-yl"
                    >
                      postTitle
                      <a
                        class="btn float-right"
                        href="/products/a/editpost/postId"
                      >
                        ...
                      </a>
                    </div>
                    <div
                      class="card-text"
                    >
                      <p>
                        postContent
                      </p>
                      

                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
